addons/addon-base-raas/packages/base-raas-services/lib/data-source/access-strategy/legacy/environment-resource-service.js

```diff
@@ -111,7 +111,7 @@ class EnvironmentResourceService extends Service {
           item: environmentScEntity.id,
         });
 
-        if (usage.added && _.size(usage.items) === 1) studiesToAdd.push(study);
```

Comment:
If we remove `usage.added` and `_.size(usage.items) === 1` from the if statement from here, it need to be checked in `addToBucketPolicy` that the policy was not already added, or else we'll run into the situation where the same policy was added multiple times, right? Same goes for other places where if statement was updated, check need to be added to subsequent methods it calls to make sure we don't add things that already exist or delete things that are still being used.

Reply:
Both addToBucketPolicy and removeFromBucketPolicy generate statements from scratch, and both rely on updateS3BucketPolicy (all 3 are idempotent on their own). This ensures multiple calls with the same params during environment provisioning/termination proceed with similar AWS service calls each time. This is therefore protecting against erroneous AWS service responses, and providing opportunity for retries.

Reply:
Got it! That makes sense. Can we remove the if statement altogether then? Since we just added a usage, `_.size(usage.items) >= 1` should always be true?